service/src/test/java/bio/terra/workspace/service/workspace/flight/gcp/CreateGcpContextFlightTest.java

```diff
@@ -168,14 +175,27 @@ void errorRevertsChanges() throws Exception {
     assertTrue(gcpCloudContextService.getGcpCloudContext(workspaceUuid).isEmpty());
 
     // Submit a flight class that always errors.
-    Map<String, StepStatus> retrySteps = getStepNameToStepStatusMap();
+    Map<String, StepStatus> undoRetrySteps = getStepNameToStepStatusMap();
+    // Cause a failure on the penultimate step. Cannot fail after the final step,
```

Comment:
i don't understand this comment about cannot fail after the final step. Isn't waitforprojectPermissionsStep that final step of cloudcontext creation?

Reply:
The final step is the common CreateContextFinishStep. Once the state change is made to READY, it is not safe to undo it, so the flight will generate a dismal failure